slimserver: 7.9.2 -> 8.3.1

jecaro opened the pull request:
Slimserver aka LogitechMediaServer was broken (see #178494)

This PR brings it to the latest version making the package usable again. I'm currently using this on nixOS running on a RPI4 and it's been working fine for the last few weeks.

I added myself as a maintainer as I'd be happy to fix the package if any new issue arises.

###### Description of changes

- Fix #178494
- upstream [changlog](https://htmlpreview.github.io/?https://github.com/Logitech/slimserver/blob/4f40dde1fdb0cc38e361e0a28e599c8e42e54ecc/Changelog8.html)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

posch commented:
Thanks for this pull request. I've tested it with Squeezebox Boom and Squeezebox Touch, and it seems to work just fine.
FWIW, I think this should be merged.

jecaro commented:
Gently pinging @phile314 

That PR has been sitting here for a few weeks now. It would be great if you could have a look soon. 

Thx in advance :pray:

phile314 commented:
As I am no longer using slimserver, I will remove myself as maintainer. I am of course more than happy if you want to take over :-)

If I have seen this correctly, this updates slimserver and uses the bundled CPAN copies from slimserver instead of the ones in nixpkgs. During the initial packaging using the bundled CPAN libraries caused trouble, but if it works nowadays that might be the easier path.

jecaro commented on pkgs/servers/slimserver/default.nix:
```diff
@@ -1,105 +1,43 @@
-    perlPackages.YAMLLibYAML
-  ];
 
+  buildInputs = [ perl534Packages.IOSocketSSL ];
```
Yes, I have read that issue when trying to package it using the modules shipped with NixOS. I found it super hard to make it works reliably that way.

It is not necessary to add a comment here as it is the way slimserver is packaged in other distros anyway (I've checked the ubuntu package and the archlinux AUR). So we just follow the recommended way to package it basically. Also, the change is clearly stated in the body of the commit message. So it'll be easily found by anybody digging into git history.

jecaro committed:
slimserver: 7.9.2 -> 8.3.1

now use the CPAN modules shipped with slimserver, this should ease the
maintainability of the package

also fix #178494

jecaro commented:
The branch has been rebased to solve the conflict introduced by https://github.com/NixOS/nixpkgs/pull/222249

adamcstephens commented:
Trying this out, but unable to install the Shairtunes2 plugin.

```
Mar 23 19:12:37 bank slimserver.pl[18459]: [23-03-23 19:12:36.6923] Slim::Utils::Misc::msg (1325) Warning: [19:12:36.6919] Can't locate auto/Crypt/OpenSSL/RSA/autosplit.ix in @INC (@INC contains: /data/lms/cache/InstalledPlugins /data/lms/cache/InstalledPlugins /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi/auto /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34.1/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34.1/x86_64-linux-thread-multi/auto /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi/auto /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34 /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/lib /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1 /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1 /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/lib/perl5/site_perl /nix/store/hywj1knkyl64ijjyqmchxvgv6nz2kdc2-perl5.34.1-Net-SSLeay-1.92/lib/perl5/site_perl/5.34.1/x86_64-linux-thread-multi /nix/store/hywj1knkyl64ijjyqmchxvgv6nz2kdc2-perl5.34.1-Net-SSLeay-1.92/lib/perl5/site_perl/5.34.1 /nix/store/hywj1knkyl64ijjyqmchxvgv6nz2kdc2-perl5.34.1-Net-SSLeay-1.92/lib/perl5/site_perl /nix/store/89vnj4nzcg22z8v0lk4rjp3wa7xjl0ni-perl5.34.1-Mozilla-CA-20200520/lib/perl5/site_perl/5.34.1/x86_64-linux-thread-multi /nix/store/89vnj4nzcg22z8v0lk4rjp3wa7xjl0ni-perl5.34.1-Mozilla-CA-20200520/lib/perl5/site_perl/5.34.1 /nix/store/89vnj4nzcg22z8v0lk4rjp3wa7xjl0ni-perl5.34.1-Mozilla-CA-20200520/lib/perl5/site_perl /nix/store/ix4jiqhghb4qaf3aiinyvw96b8fh758y-perl5.34.1-IO-Socket-SSL-2.068/lib/perl5/site_perl/5.34.1/x86_64-linux-thread-multi /nix/store/ix4jiqhghb4qaf3aiinyvw96b8fh758y-perl5.34.1-IO-Socket-SSL-2.068/lib/perl5/site_perl/5.34.1 /nix/store/ix4jiqhghb4qaf3aiinyvw96b8fh758y-perl5.34.1-IO-Socket-SSL-2.068/lib/perl5/site_perl /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/site_perl/5.34.1/x86_64-linux-thread-multi /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/site_perl/5.34.1 /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/site_perl /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/site_perl -I/nix/store/ix4jiqhghb4qaf3aiinyvw96b8fh758y-perl5.34.1-IO-Socket-SSL-2.068/lib/perl5/site_perl -I/nix/store/ /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34/x86_64-linux-thread-multi/auto /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34.1/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34.1/x86_64-linux-thread-multi/auto /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/x86_64-linux-thread-multi /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN/arch/5.34 /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/lib /nix/store/6l9rvr8wvhmf1yf465cf4366dsndw3aq-perl5.34.1-slimserver-8.3.1/CPAN /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/5.34.1/x86_64-linux-thread-multi /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/5.34.1 /data/lms/cache/InstalledPlugins/Plugins/ShairTunes2W/elib /data/lms/cache/InstalledPlugins/Plugins/ShairTunes2W/elib/5.34 /data/lms/cache/InstalledPlugins/Plugins/ShairTunes2W/elib/5.34/x86_64-linux-thread-multi /data/lms/cache/InstalledPlugins/Plugins/ShairTunes2W/elib/5.34/x86_64-linux-thread-multi/auto) at /nix/store/bgpn8pkjshbm03iy58m8g96x0aga3nci-perl-5.34.1/lib/perl5/5.34.1/AutoLoader.pm line 181.
Mar 23 19:12:37 bank slimserver.pl[18459]:  at /data/lms/cache/InstalledPlugins/Plugins/ShairTunes2W/elib/Crypt/OpenSSL/RSA.pm line 11.
```

jecaro committed:
slimserver: fix load of plugin ShairTunes2

jecaro commented:
I've added the package `perl534Packages.CryptOpenSSLRSA` in the `buildInputs`. Now the Shairtunes2 plugin seems to load fine. I don't know how to test that it works correctly thus. Do I need an Apple product for this? If so, unfortunately, I don't have one. So please try again @adamcstephens and lemme know how it goes.

As for the installed packages in the dockerfile,  `perl534Packages.CryptOpenSSLRSA` should indeed be the equivalent package to `libcrypt-openssl-rsa-perl` so I think we should be fine here. Thx for the link btw, that's a good hint of what we need to make that package works.

adamcstephens commented:
> I don't know how to test that it works correctly thus. Do I need an Apple product for this? If so, unfortunately, I don't have one. So please try again @adamcstephens and lemme know how it goes.

There may be an open source sender, but I've never looked. I updated with your changes and am now able to successfully airplay from my iphone. :+1: Thanks!

jecaro committed:
slimserver: add adamcstephens as a maintainer

jecaro commented:
>  I'm happy to help maintain if you're open to it. Feel free to add me to the list, or I can do so later.

Sure np. Happy to have more maintainers for this. Doing it now.

adamcstephens reviewed (approved):
approving again.
